src/functions.c

```diff
@@ -372,6 +375,147 @@ void fcallroCommand(client *c) {
     fcallCommandGeneric(c, 1);
 }
 
+/*
+ * FUNCTION DUMP
+ *
+ * Returns a binary blob representing all the functions.
+ * Can be loaded using FUNCTION RESTORE
+ */
+void functionDumpCommand(client *c) {
+    unsigned char buf[2];
+    uint64_t crc;
+    rio payload;
+    rioInitWithBuffer(&payload, sdsempty());
+
+    functionsSaveRio(&payload);
+
+    /* RDB version */
+    buf[0] = RDB_VERSION & 0xff;
+    buf[1] = (RDB_VERSION >> 8) & 0xff;
+    payload.io.buffer.ptr = sdscatlen(payload.io.buffer.ptr, buf, 2);
+
+    /* CRC64 */
+    crc = crc64(0, (unsigned char*) payload.io.buffer.ptr,
+                sdslen(payload.io.buffer.ptr));
+    memrev64ifbe(&crc);
+    payload.io.buffer.ptr = sdscatlen(payload.io.buffer.ptr, &crc, 8);
+
+    addReplyBulkSds(c, payload.io.buffer.ptr);
+}
+
+/*
+ * FUNCTION RESTORE <blob> [FLUSH|APPEND|REPLACE]
```

Comment:
Wouldn't it make more sense to have the blob as the last argument? That will work better with `redis-cli -x` and may also be useful if we'll handle streaming of arguments in the future.

It should not be hard to disambiguate any of these keywords from an actual blob.

Reply:
having the blob as last argument means that either the modifier args are non-optional, or they're optional, and we know they're not the blob just because the blob is mandatory and there are more args.
kinda reminds me of BLPOP where the timeout is the last arg, and anything before it is keys. i think it's bad.

not sure we should define our commands by the limitations of redis-cli, which are in any case also limiting RESTORE (if the REPLACE argument is provided).
instead, we should improve redis-cli (will also help EVAL, RESTORE, etc)

@MeirShpilraien btw, in RESTORE it's called `serialized-value` in the docs, and `payload` in the code, maybe we should fallback to the same terminology.

Reply:
> btw, in RESTORE it's called serialized-value in the docs, and payload in the code, maybe we should fallback to the same terminology.

Will change the terminology.

Regarding the payload as last argument, @oranagra I agree `redis-cli -x` should not come as a consideration but maybe streaming (in the future) should?

Reply:
Even in streaming, I. E. Handling that payload via some background operation without blocking the entire server, we need to apply thee change to redis only when the streaming ends. 
I. E. We can't flush the existing functions right away,, we must flush them only when we're done parsing the new ones, and we apply that command atomically. 
So I don't see any argument about that with regards to argument order.

Reply:
@oranagra I agree with the final point. I don't think `redis-cli` should be a consideration either, so I'm OK with the current command structure.

Reply:
> btw, in RESTORE it's called serialized-value in the docs, and payload in the code, maybe we should fallback to the same terminology.

@oranagra done